bt_ddos_shield/miner_shield.py

```diff
@@ -45,30 +50,221 @@ def __init__(self, validators_manager: AbstractValidatorsManager, address_manage
+        """
+
+        # get current state and recently fetched validators
+        current_state = self.state_manager.get_state()
+        fetched_validators: dict[Hotkey, str] = self.validators_manager.get_validators()
+
+        # remove banned validators from fetched validators
+        for banned_validator in current_state.banned_validators.keys():
+            fetched_validators.pop(banned_validator, None)
+
+        # calculate difference between current state and fetched validators
+        deprecated_validators = current_state.known_validators.keys() - fetched_validators.keys()
+        new_validators = fetched_validators.keys() - current_state.known_validators.keys()
+        changed_validators = {
+            k: fetched_validators[k] for k in fetched_validators.keys() & current_state.known_validators.keys()
+                if fetched_validators[k] != current_state.known_validators[k]
+        }
+
+        # handle changes in validators
+
+        self.event_processor.add_event(MinerShieldEvent(
+            f"Handling validators change, deprecated_validators count={len(deprecated_validators)}"
+            f", new_validators count={len(new_validators)}, changed_validators count={len(changed_validators)}")
+        )
+
+        for validator in deprecated_validators:
+            self.event_processor.add_event(MinerShieldEvent(f"Removing validator {validator}"))
+
+            if validator in current_state.active_addresses:
+                self.address_manager.remove_address(current_state.active_addresses[validator])
+
+            self.state_manager.remove_validator(validator)
+
+        # TODO handle new_validators and changed_validators
+
+        if deprecated_validators or new_validators or changed_validators:
+            # if anything changed update manifest file and publish new version to blockchain
+            # TODO also check state of shield if manifest was published at all
+            pass
+
+    def _handle_ban_validator(self, validator_hotkey: Hotkey):
+        """
+        Ban validator by its hotkey. It will update manifest file and publish info about new file version to blockchain.
+        """
+        # TODO
         pass
+
+
+class MinerShieldTask:
+    """
+    Task to be executed by shield worker.
+    """
+
+    def __init__(self, task_name: str):
+        """
+        Initialize task.
+
+        Args:
+            task_name: Short name of the task.
+        """
+        self.task_name = task_name
+
+    def handle(self, miner_shield: MinerShield):
+        """
+        Run task logic.
+
+        Args
+            miner_shield: Instance of MinerShield in which task is executed.
+        """
+        pass
+
+    def __repr__(self):
+        return self.task_name
+
+class MinerShieldInitializeTask(MinerShieldTask):
+    def __init__(self):
+        super().__init__("INITIALIZE_SHIELD")
+
+    def handle(self, miner_shield: MinerShield):
+        miner_shield._handle_initialize()
```

Comment:
This could be refactored away into the abstract class too, but it's not worth it - we wouldn't save much anyway and it'd confuse static code analyzers